src/jemalloc.c

```diff
@@ -2685,6 +2685,67 @@ je_free(void *ptr) {
 		}
 		check_entry_exit_locking(tsd_tsdn(tsd));
 	}
+}
+
+JEMALLOC_ALWAYS_INLINE
+bool free_fastpath(void *ptr, size_t size, bool size_hint) {
+	tsd_t *tsd = tsd_get(false);
+	if (unlikely(!tsd || !tsd_fast(tsd))) {
+		return false;
+	}
+
+	tcache_t *tcache = tsd_tcachep_get(tsd);
+
+	alloc_ctx_t alloc_ctx;
+	if (!size_hint || config_cache_oblivious) {
```

Comment:
Why is cache_oblivious involved here? I don't really follow.

Reply:
I'll add a comment

Similar to isfree() - if cache_oblivious is *off*, we can check for sampled objects by checking Page alignment.  If we are cache_oblivious, the start address is randomized, and we can't do that.

                /*                                                                                                                                                                                                                       
                 * When cache_oblivious is disabled and ptr is not page aligned,                                                                                                                                                         
                 * the allocation was not sampled -- usize can be used to                                                                                                                                                                
                 * determine szind directly.                                                                                                                                                                                             
                 */